extension-manifest-v3/src/background/adblocker.js

```diff
@@ -326,7 +339,10 @@ if (__PLATFORM__ === 'safari') {
   });
 } else {
   chrome.webNavigation.onCommitted.addListener(async (details) => {
-    injectScriptlets(details.tabId, details.url);
+    const isMainFrame = details.parentFrameId === -1;
+    if (isMainFrame || !(await isPausedTab(details.tabId))) {
```

Comment:
I have an idea to use `tabStats` here synchroniusly. As the mainframe must call this event before subframes, we can find stats for the `details.tabId` and have the `stats.hostname` here.

Reply:
```suggestion
    if (isMainFrame || !(await isPaused(tabStats.get(details.tabId)?.hostname) {
```

Reply:
This is a follow-up that tries out the idea: https://github.com/ghostery/ghostery-extension/pull/1692